Sources/FoundationEssentials/Data/Data.swift

```diff
@@ -63,8 +63,9 @@ internal func _withStackOrHeapBuffer(capacity: Int, _ body: (UnsafeMutableBuffer
         )
         withUnsafeMutableBytes(of: &buffer) { buffer in
             assert(buffer.count == inlineCount)
-            let start = buffer.baseAddress!.assumingMemoryBound(to: UInt8.self)
-            body(UnsafeMutableBufferPointer(start: start, count: capacity))
+            buffer.withMemoryRebound(to: UInt8.self) {
```

Comment:
@atrick suggested the assuming variant; does this no longer force the buffer to be permanently bound?

Reply:
In this spot we could use `assuming`, since the provenance is clear. The temporary binding variant reverts the previous state, so I think it ends up with the same states.